diff --git a/dist/index.mjs b/dist/index.mjs
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -206,11 +206,17 @@
 //#region src/utils/format/cleanse-ansi.constants.ts
 const CLEAR_LINE_REGEX = "(?:\\u001b|\\u009b)\\[[\\=><~/#&.:=?%@~_-]*[0-9]*[\\a-ln-tqyz=><~/#&.:=?%@~_-]+";
 const BELL_REGEX = /\u0007/;
+const HYPERLINK_REGEX = /(?:\u001b\]8;;.*?\u0007.*?\u001b\]8;;\u0007)/g;
 
 //#endregion
 //#region src/utils/format/cleanse-ansi.ts
 function cleanseAnsi(chunk) {
-	return String(chunk).replace(new RegExp(CLEAR_LINE_REGEX, "gmi"), "").replace(new RegExp(BELL_REGEX, "gmi"), "").trim();
+	const hyperlinks = chunk.match(HYPERLINK_REGEX) || [];
+	let cleansedChunk = String(chunk).replace(new RegExp(HYPERLINK_REGEX, "g"), "__HYPERLINK__").replace(new RegExp(CLEAR_LINE_REGEX, "gmi"), "").replace(new RegExp(BELL_REGEX, "gmi"), "").trim();
+	hyperlinks.forEach((link) => {
+		cleansedChunk = cleansedChunk.replace("__HYPERLINK__", link);
+	});
+	return cleansedChunk;
 }
 
 //#endregion
@@ -2092,10 +2098,11 @@
 		if (!await task.check(this.ctx)) return;
 		return new TaskWrapper(task).run(this.ctx);
 	}
-	signalHandler() {
+	async signalHandler() {
 		this.tasks?.forEach(async (task) => {
 			if (task.isPending()) task.state$ = ListrTaskState.FAILED;
 		});
+		await this.externalSignalHandler?.(this.ctx);
 		if (this.isRoot()) {
 			this.renderer?.end(/* @__PURE__ */ new Error("Interrupted."));
 			process.exit(127);
